charts/spire/charts/spire-server/templates/deployment.yaml

```diff
@@ -0,0 +1,472 @@
+{{- $configSum := (include (print $.Template.BasePath "/configmap.yaml") . | sha256sum) }}
+{{- $configSum2 := (include (print $.Template.BasePath "/secret.yaml") . | sha256sum) }}
+{{- $configSum3 := (include (print $.Template.BasePath "/controller-manager-configmap.yaml") . | sha256sum) }}
+{{- $configSumTornjak := (include (print $.Template.BasePath "/tornjak-config.yaml") . | sha256sum) }}
+{{- $fullname := include "spire-server.fullname" . }}
+
+{{- if eq .Values.deploymentType "deployment" }}
+{{- if (has .Values.persistence.type (list "pvc" "hostPath")) }}
+{{- fail "When running as deployment, persistence can't be set. 'persistence.type' must be [\"emptyDir\"]" }}
+{{- end }}
+{{- if (eq .Values.dataStore.sql.databaseType "sqlite3") }}
+{{- fail "When running as deployment, sqlite3 can't be used." }}
+{{- end }}
+{{- if (eq (.Values.keyManager.disk.enabled | toString) "true") }}
+{{- fail "When running as deployment, disk keymanager can't be used. 'keyManager.disk.enabled' must be false." }}
+{{- end }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "spire-server.fullname" . }}
+  namespace: {{ include "spire-server.namespace" . }}
+  labels:
+    {{- include "spire-server.labels" . | nindent 4 }}
+    app.kubernetes.io/component: server
+
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "spire-server.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: server
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: spire-server
+        checksum/config: {{ $configSum }}
+        checksum/config2: {{ $configSum2 }}
+        checksum/config3: {{ $configSum3 }}
+        checksum/configTornjak: {{ $configSumTornjak }}
+        {{- with .Values.podAnnotations }}
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "spire-server.selectorLabels" . | nindent 8 }}
+        app.kubernetes.io/component: server
+        component: server
+        release: {{ .Release.Name }}
+        release-namespace: {{ .Release.Namespace }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
```

Comment:
There is a large amount of duplication here which will make this much harder to maintain...

Can we make all the stuff in the statefulset.spec.template.spec and deployment.spec.tempate.spec into a new file,
`_pod_spec.tpl` wrapped in:
```
{{ define "spire-server.pod-spec" }}
...
{{ end }}
```

and included here and in the statefulset with something like
```
{{ include "spire-server.pod-spec" . | nindent 6 }}
```

Reply:
Moved spec to _pod_spec.